app/sections/collection-header.tsx

```diff
@@ -0,0 +1,142 @@
+import type {
+    HydrogenComponentProps,
+    HydrogenComponentSchema,
+} from '@weaverse/hydrogen';
+import { forwardRef } from 'react';
+import { useLoaderData } from '@remix-run/react';
+import { CollectionDetailsQuery } from 'storefrontapi.generated';
+
+interface HeaderProps extends HydrogenComponentProps {
+    heightSectionDesktop: number;
+    heightSectionMobile: number;
+    enableBackground: boolean;
+    textPosition: string;
+    overlayBackground: number;
+    enableOverlay: boolean;
+}
+
+let CollectHeader = forwardRef<HTMLElement, HeaderProps>((props, ref) => {
+    let { heightSectionDesktop, heightSectionMobile, enableBackground, textPosition, overlayBackground, enableOverlay, ...rest } = props;
+    let { collection } = useLoaderData<
+        CollectionDetailsQuery & {
+            collections: Array<{ handle: string; title: string }>;
+        }
+    >();
+    let backgroundStyle = {};
```

Comment:
I think **Typescript** would cry here, please add the type for all of your `style` object with the value `CSSProperties` that imported from `react